CRM/Utils/Cache/Redis.php

```diff
@@ -161,7 +164,9 @@ public function flush() {
     // more general rethink of cache expiration/TTL.
 
     $keys = $this->_cache->keys($this->_prefix . '*');
-    $this->_cache->del($keys);
+    if ( $keys !== false ) {
```

Comment:
like above, if keys returns false, then del(false) will fail due to prefix based ACL permissions